portofino-core/src/main/java/com/manydesigns/portofino/spring/PortofinoSpringConfiguration.java

```diff
@@ -55,6 +59,23 @@
 
     @Bean(name = DEFAULT_BLOB_MANAGER)
     public BlobManager getDefaultBlobManager() {
+        if(configuration.getString(PortofinoProperties.BLOB_MANAGER_TYPE, "standard").equalsIgnoreCase( "s3" )){
+            String region = configuration.getString(PortofinoProperties.AWS_REGION);
+            String bucketName = configuration.getString(PortofinoProperties.AWS_S3_BUCKET);
+            String location =   configuration.getString(PortofinoProperties.AWS_S3_LOCATION);
+            logger.info("Using S3 blob manager");
+            if(StringUtils.trimToNull( region )==null){
+                logger.error(PortofinoProperties.AWS_REGION+ " property not found" );
+            }
+            if(StringUtils.trimToNull( bucketName )==null){
+                logger.error( PortofinoProperties.AWS_S3_BUCKET + " property not found" );
+            }
+            if(StringUtils.trimToNull(location) == null)
+                return new S3BlobManager(region, bucketName);
+            else
+                return new S3BlobManager(region, bucketName, location);
+        }
+
```

Comment:
Ipotizzando di mettere il blob manager in un modulo portofino-aws, c'è il problema che portofino-core non può dipendere da tale modulo.
Allora io rifattorizzerei questa parte in una BlobManagerFactory che si può configurare nel portofino.properties, per es.: 
`blobmanager.factory = com.manydesigns.portofino.aws.s3.S3BlobManagerFactory`
A questo punto getDefaultBlobManager controlla se è specificata una factory, nel caso la istanzia (passandole la configurazione, l'application directory, e tutto quello che può servire) e la invoca per ottenere il blob manager. Altrimenti, esegue la logica standard (che potrebbe a questo punto essere spostata in una BlobManagerFactory di default in portofino-core).

Reply:
Va bene, procedo con il creare un modulo per aws e la BlobManagerFactory;
invece di cercare nel portofino.properties non potremmo sfruttare la DI di spring? 
Rendo la BlobManagerFactory un bean e creo una DefaultBlobManagerFactory si comporta come `PortofinoSpringConfiguration.getDefaultBlobManager` e getDefaultBlobManager dipende dalla factory e chiama `factory.getBlobManager()`?

Reply:
Sì sarebbe meglio usare la DI di Spring ma bisogna capire come attivare il blob manager aws invece di quello di default.